src/modules/common/hocs/withLanguageNotAvailable.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+
+import * as React from 'react'
+import { LanguageModel } from '@integreat-app/integreat-api-client'
+import LanguageNotAvailableContainer from '../containers/LanguageNotAvailableContainer'
+
+export type LanguageNotAvailableType = {|
+  languageNotAvailable: boolean,
+  city: string,
+  languages: Array<LanguageModel>,
+  changeUnavailableLanguage: (city: string, newLanguage: string) => void
+|}
+
+type PropsType<T> = LanguageNotAvailableType | T
```

Comment:
As T is only `*` this generic is unused. Also  think you are handling the tyoe of PropsType wrong in these HOCs.

There should be examples in the webapp or in some other HOC.